EscelForce.Business.Login/Services/MapExtraction/CreateExtractionMapService.cs

```diff
@@ -65,18 +66,22 @@ public ServiceResponseModel<bool> SubmitOnObjectSelection(string objectName)
                     Objects = new List<SfObject>()
                 };
 
-                if (!query.Objects.Any(x => x.Name == objectName))
+                string selectedObjectName = objectName.Contains('|')? objectName.Split('|')[1].Trim(): objectName.Trim();
```

Comment:
Please create static methods to parse objectName and API Name in the SfObject class itself, as `GetObjectNameFromDisplayName()` & `GetApiNameFromDisplayName()`. The parsing logic is not relevant to our `CreateExtractionMapService`. Also, `.Split` would take care of handling the `Contains` operation. You can directly `Split` after performing a `null` check.

Reply:
You will have to provide exclusive checks in the methods suggested and return null if the 2nd element of the array does not exist. Check for the length of the array and proceed accordingly.

Reply:
Changed it